libyul/backends/evm/SSAControlFlowGraphBuilder.cpp

```diff
@@ -112,45 +133,44 @@ SSACFG::ValueId tryRemoveTrivialPhi(SSACFG& _cfg, SSACFG::ValueId _phi)
 				if (input == _phi)
 					uses.emplace_back(Use{std::ref(op), n});
 		}
-		block.forEachExit([&](SSACFG::BlockId _block) {
-			_recurse(_block, _recurse);
-		});
-	};
-	auto phiBlock = phiInfo->block;
-	_cfg.block(phiBlock).phis.erase(_phi);
-	findUses(phiBlock, findUses);
-
-	if (!same.hasValue())
-	{
-		// This will happen for unreachable paths.
-		// TODO: check how best to deal with this
-		same = _cfg.unreachableValue();
+		if (auto* functionReturn = std::get_if<SSACFG::BasicBlock::FunctionReturn>(&block.exit))
+			std::replace(
+				functionReturn->returnValues.begin(),
+				functionReturn->returnValues.end(),
+				_phi,
+				same
+			);
```

Comment:
(So *at least* just a direct ``std::replace`` on each operations input instead of storing where they are in ``uses`` and doing the replacement afterwards should be possible and less pathological than the original version I wrote :-))

Reply:
Lol, the lambdas resizing the visited vector, not sure what I was thinking when I wrote this :-D.

Reply:
Yeah you are right, it should be for all exits! Not sure why I didn't do that in the first place lol :D